docs/docs/core/database/relations/reordering.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: Relations
+slug: /database/relations
+description: Conceptual guide to relations reordering in the Database
+tags:
+  - database
+  - relations
+  - reordering
+---
+
+Strapi allows to reorder a relation list.
+
+<img src="/img/database/reordering.png" alt="An example of reordering in the CM" />
+
+This reordering feature is available in the Content Manager and the API.
+
+## Code location
+
+`packages/core/database/lib/entity-manager/relations-orderer.js`
+
+## How order is stored in DB?
+
+- We store the order of the relations in a `order` field.
+- For bidirectional relations, we store the order of the other side in a `inverse_order` field.
+
+We store order for all type of relations, except for:
+
+- Polymorphic relations (too complicated to implement).
+- One to one relations. (as there is only one relation per pair)
+
+### Many to many (Addresses <-> Categories)
+
+<img src="/img/database/m2m-example.png" alt="many to many relation" />
+
+- `category_order` is the order of the categories relations in an address entity.
+- `address_order` is the order of the addresses relations in a category entity.
+
+### One to one (Kitchensinks <-> Tags)
+
+<img src="/img/database/o2o-example.png" alt="one to one relation" />
+
+- there is no `order` fields as there is only one relation per pair.
+
+### One way relation (Restaurants <-> Categories)
+
+Where a restaurant has many categories:
+
+<img src="/img/database/mw-example.png" alt="many way relation" />
+
+- `category_order` is the order of the categories relations in a restaurant entity.
+- There is no `restaurant_order` as it is a one way relation.
+
+## How to use this in the DB layer?
```

Comment:
```suggestion
## How to reorder relations in the DB layer
```

Personally I try to avoid using `this` because `this` can be anything and it's better to be specific :)